debian/rules

```diff
@@ -11,3 +11,6 @@ override_dh_auto_test:
 override_dh_install:
 	mkdir -p debian/docker-push-latest-if-changed/usr/bin
 	cp docker_push_latest_if_changed.py debian/docker-push-latest-if-changed/usr/bin/docker-push-latest-if-changed
+
+override_dh_builddeb:
+	dh_builddeb -- -Zgzip
```

Comment:
this shouldn't be necessary afaik -- this is probably an indication that you're building on an incompatible debian version and then trying to install on an older version.

debian changed from gzip to xz in recent versions of `dpkg` -- you probably need to either build on older debian or build separate packages.

this hack "appears" to make it work but you're probably going to hit other breakage since dpkg changed quite a few other internals between versions

Reply:
in that case, build on the oldest one you support?

Reply:
Whelp made it build individual `.deb` files by version and bionic can't even install its own `control.tar.xz`d up packages without installing `build-essential` or at least something in it

https://gist.github.com/ATRAN2/c8dd0b2f44a8ed2447ebb5b3cb7a1806

Reply:
`build-essential` is a giant package, probably not actually helpful in figuring out which dep is missing

Reply:
Alright it's `xz-utils` that needs to be installed https://gist.github.com/ATRAN2/c26d13d9de888f7eb3970a704766ec86